apps/condo/domains/common/components/tasks/TaskProgress.tsx

```diff
@@ -141,13 +150,24 @@ export const TaskProgressTracker: React.FC<ITaskProgressTrackerProps> = ({ task
         }
     })
 
-    const handleRemoveTask = useCallback(() => {
+    const removeTaskFromUI = () => {
         if (removeStrategy.includes(TASK_REMOVE_STRATEGY.STORAGE)) {
             removeTaskFromStorage(record)
         } else if (removeStrategy.includes(TASK_REMOVE_STRATEGY.PANEL)) {
             deleteTaskFromContext(record)
         }
-    }, [record, storage, deleteTaskFromContext, removeTaskFromStorage])
+    }
+
+    const cancelTaskAndRemoveFromUI = storage.useUpdateTask({ status: TASK_STATUS.CANCELLED }, removeTaskFromUI)
+
+    const handleRemoveTask = useCallback(async () => {
+        // Tasks under processing should be cancelled before removing from UI
+        if (record.status === TASK_STATUS.PROCESSING) {
+            cancelTaskAndRemoveFromUI({}, record)
+        } else {
+            removeTaskFromUI()
+        }
+    }, [record, storage, removeTaskFromUI, deleteTaskFromContext])
```

Comment:
`storage` and `deleteTaskFromContext` aren't used within the function, so it's better to remove them from the dependencies to prevent unnecessary changing of `handleRemoveTask `.